static/js/src/advantage/api/contracts-types.ts

```diff
@@ -0,0 +1,1694 @@
+/**
+ * This file was largely auto-generated by openapi-typescript.
+ * It may be easiest to regenerated this file from the contracts API definitions
+ * rather than updating this file directly.
+ * This file should remain consistent with the contracts API, any types that
+ * need to be augmented should be done in types.ts.
+ */
```

Comment:
@ClementChaumel these types will probably help you when/if you start migrating your components.

My approach is to define these types exactly as they come from the contracts API and then in the types.ts file I'm defining types that include any additional data that gets added in the view.py.